cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/QueryResult.java

```diff
@@ -122,10 +122,16 @@ public DocumentRevision next() {
             if (!this.hasNext()) {
                 throw new NoSuchElementException();
             }
-            if (!documentBlock.hasNext()) {
+
+            DocumentRevision doc =  documentBlock.next();
```

Comment:
thoughts @ricellis 
```java
            // it is always safe to call documentBlock.next() without calling hasNext because
            // the this.hasNext() will return false if there are no more documents to read this
            // is because we load the next block of documents (if there are any), before
            // returning the last document from the last batch, ensuring that documentBlock.next()
            // will always have a document to return.
            DocumentRevision doc =  documentBlock.next();

```